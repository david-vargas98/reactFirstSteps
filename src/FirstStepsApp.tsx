import { ItemCounter } from "./shopping-cart/ItemCounter";

export function FirstStepsApp() {
    return (
        <>
            <h1>Shopping cart</h1>

            <ItemCounter name="Nintendo switch 2" quantity={1} />
            <ItemCounter name="Pro controller" quantity={1} />
            <ItemCounter name="Super smash" quantity={1} />
        </>
    )
}