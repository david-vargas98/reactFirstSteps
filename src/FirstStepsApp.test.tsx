import { describe, expect, test } from 'vitest';
import { render ,screen } from '@testing-library/react';

import { FirstStepsApp } from './FirstStepsApp';

// Challenge: make the snapshot from the component
describe('FirstStepsApp', () => {
    test('should match snapshot', () => { //instructor says won't show this test anymore for the rest of the course
        render(<FirstStepsApp />);

        expect(screen.getAllByRole('root')).toMatchSnapshot();
    })
});