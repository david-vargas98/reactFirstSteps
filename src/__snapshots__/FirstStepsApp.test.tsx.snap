// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FirstStepsApp > should match snapshot 1`] = `
[
  <div
    role="root"
  >
    <h1>
      Shopping cart
    </h1>
    <section
      class="_item-row_8b32a2"
    >
      <span
        class="_itemText_8b32a2"
        style="color: red;"
      >
        Nintendo switch 2
      </span>
      <button>
        +1
      </button>
      <span>
        1
      </span>
      <button>
        -1
      </button>
    </section>
    <section
      class="_item-row_8b32a2"
    >
      <span
        class="_itemText_8b32a2"
        style="color: black;"
      >
        Pro controller
      </span>
      <button>
        +1
      </button>
      <span>
        3
      </span>
      <button>
        -1
      </button>
    </section>
    <section
      class="_item-row_8b32a2"
    >
      <span
        class="_itemText_8b32a2"
        style="color: black;"
      >
        Super smash
      </span>
      <button>
        +1
      </button>
      <span>
        5
      </span>
      <button>
        -1
      </button>
    </section>
  </div>,
]
`;
